servers/services/src/main/java/org/projectnessie/services/rest/MultiTenantFilter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.services.rest;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.ext.Provider;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.projectnessie.model.Validation;
+
+/**
+ * Extracts the optional repository owner+name URI path parts and configured the request-scoped
+ * {@link MultiTenant} bean accordingly.
+ *
+ * <p>Non-multi-tenant URIs just do not have owner+repo information in the URI.
+ *
+ * <p>Multi-tenant URIs have the owner+repo parts after the {@code /api/v1} URI path part, for
+ * example: {@code http://localhost:19120/api/v1/owner/repo/trees/tree}.
+ */
+@PreMatching
+@Provider
+public class MultiTenantFilter implements ContainerRequestFilter {
+
+  private final Pattern multiTenantPathPattern;
+  private final String restPath;
+
+  @Inject MultiTenant multiTenant;
+
+  // TODO @ConfigProperty is the way in Quarkus to get the REST-API base path, but it's probabyl
+  //  not the right thing in Jersey. It would also be much nicer to have the owner+repo part
+  //  before the /api/v1 URI part.
+  @Inject
+  public MultiTenantFilter(@ConfigProperty(name = "quarkus.resteasy.path") String restPath) {
```

Comment:
I don't think this is compatible w/ all impls so we should prob take care of it now given where jersey PR is at

Reply:
Will update